tests/test_h1_client.c

```diff
@@ -1553,10 +1553,43 @@ H1_CLIENT_TEST_CASE(h1_client_close_from_on_thread_makes_not_open) {
     return AWS_OP_SUCCESS;
 }
 
-/* Send "Connection: Upgrade" request and receive "101 Switching Protocols" response. */
-static int s_switch_protocols(struct tester *tester) {
-    /* send request */
-    struct aws_http_header headers[] = {
+struct protocol_switcher {
+    /* Settings */
+    struct tester *tester;
+    size_t downstream_handler_window_size;
+    const char *data_after_upgrade_response;
+    bool dont_install_downstream_handler;
```

Comment:
negative bools :( Perhaps ```needs_downstream_handler```, or ```should_install_downstream_handler```, or keep it flipped, but something like ```no_downstream_handler``` or ```is_terminal_handler```

Reply:
ok